vipps-login-api-checklist.md

```diff
@@ -11,46 +11,50 @@ pagination_prev: null
+| Do not use embedded iFrames | Integration with the Vipps Login API is redirect-based (i.e., no embedded iFrame). |
+|  Set company name and logo  | The sales unit name appears on the Vipps Login landing page. Both the name and logo appear in the Vipps app under _Personal information_ > _Companies with access_. See [how to change name and logo](vipps-login-api-faq.md#how-can-i-change-my-name-and-logo).|
+|   Whitelist redirect URIs   | Ensure that all `redirect_uri`s for your integration have been added using to the merchant portal white list. See the [FAQ](vipps-login-api-faq.md#how-can-i-activate-and-set-up-vipps-login) for how to do this. Ensure that all URLs use `HTTPS` or native URL schemes (i.e., `vipps://`), not `HTTP`. ALL URLs must match the exact URI sent on `/auth` request. No query parameters or additional trailing `/` can be included. |
+|  Use only required scopes | Request only the scopes that you require, no "nice to have" scopes. If using national identity number, you must be granted access to request these scopes. See [FAQ](vipps-login-api-faq.md#who-can-get-access-to-nin-and-how) for details. |
+|  Comply with Vipps terms and conditions   | If you, as a merchant, will act on behalf of others (share data you have gotten from Vipps with other merchants), ensure that you comply with our terms and conditions to do this. This is shown in section 8.4 in our [terms and conditions for merchants](api-guide/overview.md#vipps-login-from-phone-number). |
+|    Present terms and conditions | Terms and conditions are presented to the user, and the necessary consents are collected from the user (i.e., consent to marketing purposes, etc.). |
+
+
+## Security
+
+| Action | Comment |
+|--------|---------|
+| Generate a `state` parameter    | Ensure that a unique [`state` parameter](vipps-login-api-faq.md#whats-the-purpose-of-the-state-parameter) is generated for each `/auth` request. |
+| Do not share `client_secret`   | Ensure that the `client_secret` is kept secret and is never be shared to browsers or native apps. |
+
+
+## Quality assurance
+
+| Action | Comment |
+|--------|---------|
+|  Ensure functionality   | Ensure that your solution is verified to work if the user start in a "non-default" browser on mobile, e.g. start Vipps Login from Chrome browser on iOS. |
+| Update user registry    | Implement proper linking of the Vipps user to your own user registry. This login must be based on either phone number or e-mail address. See [recommendations on linking to user account](api-guide/important-information.md#Recommendations-on-linking-to-user-account). |
+|     Handle errors | Make sure to log and handle all errors. Cancelled logins and error situations on the redirect back to `redirect_uri` (redirects with a `error` query parameter) are handled. All integrations should to display errors in a way that the users (customers and merchant employees/administrators) can see and understand them.|
```

Comment:
Something strange with these sentences
```
|     Handle errors | Make sure to log and handle all errors. Cancelled logins and error situations on the redirect back to `redirect_uri` (redirects with a `error` query parameter) are handled.
```

Reply:
Yes, you're right. I will address it. :-)

Reply:
Make sure to log and handle all errors. For example, handle cancelled logins and error situations while redirecting the user back to redirect_uri (i.e., redirect with an error query parameter). In addition, display errors in a way that the users (customers and merchant employees/administrators) can see and understand them.